templates/default/etcd.initd.erb

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+#
+# etcd      This shell script takes care of starting and stopping
+#             etcd
+#
+# chkconfig: 345 80 30
+# description:  etcd service registry
+
+# processname: etcd
+# pidfile: /var/run/etcd/etcd.pid
+
+# Source function library.
+. /lib/lsb/init-functions
+
+# Source etcd configuration
+if [ -f /etc/default/etcd ] ; then
+ . /etc/default/etcd
+fi
+
+RETVAL=0
+prog="etcd"
+
+start() {
+ # Start daemons.
+ echo -n $"Starting $prog: "
+ PID=`pidof etcd`
+ if [ ! -z $PID ]; then
+   return 2
+ fi
+ /usr/local/bin/etcd <%= @arguments.join("\\\n  ") %> > /var/log/etcd 2> /var/log/etcd.err &
+ RETVAL=$?
+ echo
+ [ $RETVAL -eq 0 ] && touch /var/lock/etcd
```

Comment:
Instead of using `pidof` to find the service's process, how about writing its PIF to `/var/lock`? (I'm not familiar with RHEL's `/var/lock` conventions/constraints... `/var/pid` if writing into the lock file is a bad idea?)